controllers/sr_controller/sr_controller.py

```diff
@@ -137,27 +140,65 @@ def reconfigure_environment(robot_file: Path) -> None:
     os.chdir(str(robot_file.parent))
 
 
+def log_filename(zone_id: int) -> str:
+    now = datetime.datetime.now()
+    return 'log-zone-{}-{}.txt'.format(
+        zone_id,
+        now.isoformat().replace(':', ''),
+    )
+
+
+class TeeStdout:
+    def __init__(self, name: Path) -> None:
+        self.name = name
+
+    def __enter__(self) -> None:
+        self.file = open(str(self.name), mode='w').__enter__()
+        self.stdout = sys.stdout
+        sys.stdout = self  # type: ignore
```

Comment:
We can use the `redirect_stdout` context manager for this.

Reply:
5d62b19